ext/opentelemetry-ext-django/src/opentelemetry/ext/django/middleware.py

```diff
@@ -0,0 +1,110 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from logging import getLogger
+
+from django.utils.deprecation import MiddlewareMixin
```

Comment:
Is the middleware method of patching on the path of deprecation? Is there a more recent way of doing this?

Reply:
No, `deprecation` there does not mean that [`MiddlewareMixin`](https://docs.djangoproject.com/en/3.0/topics/http/middleware/#upgrading-pre-django-1-10-style-middleware) is to be deprecated but that it works with recent and deprecated Django versions.

Reply:
While there is no formal deprecation road for this mixin yet apparently, there is no reason at all to use this compatibility stub. It only exists to ease migrating old-style middleware to the new style. That old style was only supported until v1.11 which already is past it's EOL, and as of v2.0 ([see the last item on this page of the release notes](https://docs.djangoproject.com/en/3.0/releases/2.0/)) is officially dropped, this is no longer supported and shouldn't be used anymore. Especially since this is brand new code, with a minimum required Django version of v2.2, so already some versions after that, it's bad practice to keep dragging this legacy approach along. I made [an MR for the opencensus django integration](https://github.com/census-instrumentation/opencensus-python/pull/710) to switch to new style and remove the mixin, which got as a comment to wait until the last version still supporting was EOL. Seeing as we're way beyond that point now, this should not be done.